demos/viscoelastic/viscoelastic_loading.py

```diff
@@ -0,0 +1,517 @@
+        centre = radius_values[i] - radius_values[0]
+        mag = background_values[i] - background_values[i-1]
+        profile += step_func(depth, centre, mag, increasing=False, sharpness=sharpness)
+
+    field.interpolate(profile)
+
+
+density = Function(W, name="density")
+initialise_background_field(density, density_values)
+
+shear_modulus = Function(W, name="shear modulus")
+initialise_background_field(shear_modulus, shear_modulus_values)
+
+viscosity = Function(W, name="viscosity")
+initialise_background_field(viscosity, viscosity_values)
+# -
+
+# Next let's create a function to store our ice load. Following the long test from Weeredesteijn et al 2023, during the first 90 thousand years of the simulation the ice sheet will grow to a thickness of 1 km. The ice thickness will rapidly shrink to ice free conditons in the next 10 thousand years. Finally, the simulation will run for a further 10 thousand years to allow the system to relax towards isostatic equilibrium. This is approximately the length of an interglacial-glacial cycle. The width of the ice sheet is 100 km and we have used a tanh function again to smooth out the transition.
+
+# +
+rho_ice = 931
+g = 9.8125
+
+# Initialise ice loading
+ice_load = Function(W, name="Ice load")
+Hice = 1000
+year_in_seconds = Constant(3600 * 24 * 365.25)
+T1_load = 90e3 * year_in_seconds
+T2_load = 100e3 * year_in_seconds
+
+# Disc ice load but with a smooth transition given by a tanh profile
+disc_radius = 100e3
+disc_dx = 5e3
+k_disc = 2*pi/(8*disc_dx)  # wavenumber for disk 2pi / lambda
+r = X[0]
+disc = 0.5*(1-tanh(k_disc * (r - disc_radius)))
+ramp = Constant(0)
+
+
+# -
+
+# Next let's define the length of our time step. If we want to accurately resolve the elastic response we should choose a timestep lower than the Maxwell time, $\alpha = \eta / \mu$. The Maxwell time is the time taken for the viscous deformation to 'catch up' the initial, instantaneous elastic deformation.
+#
+# Let's print out the maxwell time for each layer
+
+for layer_visc, layer_mu in zip(viscosity_values, shear_modulus_values):
+    log(f"Maxwell time: {float(layer_visc/layer_mu/year_in_seconds):.0f} years")
+
+# As we can see the shortest Maxwell time is given by the lower mantle and is about 280 years, i.e. it will take about 280 years for the viscous deformation in that layer to catch up any instantaneous elastic deformation. Conversely the top layer, our lithosphere, has a maxwell time of 6 million years. Given that our simulations only run for 110000 years the viscous deformation over the course of the simulation will always be negligible compared with the elastic deformation. For now let's choose a timestep of 250 years and an output time step of 2000 years.
+
+# +
+# Timestepping parameters
+Tstart = 0
+time = Function(R).assign(Tstart * year_in_seconds)
+
+dt_years = 250
+dt = Constant(dt_years * year_in_seconds)
+Tend_years = 110e3
+Tend = Constant(Tend_years * year_in_seconds)
+dt_out_years = 2e3
```

Comment:
For consiustency with other demos, perhaps output_frequency_years instead of dt_out_years?